### PR TITLE
Feature/best city to saving money

### PR DESCRIPTION
**Pull request by AmeerAmjed**

### CODE REVIEW

File: src/test/kotlin/interactor/GetBestCityForSavingMoneyInteractorTest.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.TestInstance
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class GetBestCityForSavingMoneyInteractorTest {
+
+
+    lateinit var bestCity: GetBestCityForSavingMoneyInteractor
```
Comment (TarekIdrees) [inline src/test/kotlin/interactor/GetBestCityForSavingMoneyInteractorTest.kt]: bestCity should be replaced with getBestCityFor...........

File: src/test/kotlin/interactor/GetBestCityForSavingMoneyInteractorTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        bestCity = GetBestCityForSavingMoneyInteractor(FakeDataProvider)
+    }
+
+
+    @Test
+    fun should_ReturnTrue_When_TheCityHasNotNullRequireValues(){
+
+        // given a city haven't null fields
+        val cityWithoutNullFields = dataSource.getAllCitiesData()[0]
```
Comment (TarekIdrees) [inline src/test/kotlin/interactor/GetBestCityForSavingMoneyInteractorTest.kt]: real data for testing? (**dataSource.getAllCitiesData()[0]**)

File: src/main/kotlin/interactor/GetBestCityForSavingMoneyInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+class GetBestCityForSavingMoneyInteractor(private val dataSource: CostOfLivingDataSource){
+
+    fun execute(isAppartmentInTheCityCentre: Boolean): CityEntity {
+
+        isAppartmentInCenter = isAppartmentInTheCityCentre
+
+        val cities = dataSource.getAllCitiesData()
+        var bestCity = cities[0]
+
+        for (city in cities) {
```
Comment (TarekIdrees) [inline src/main/kotlin/interactor/GetBestCityForSavingMoneyInteractor.kt]: should be replaced with kotlin functions e.g( **List.forEach{} , List.filter{}** )

**Review by TarekIdrees — CHANGES_REQUESTED** [review]
changes requested

**Review by TarekIdrees — APPROVED** [review]

**Review by msalarab — APPROVED** [review]

**Pull request merged**